### PR TITLE
Integrating artifacts

### PR DESCRIPTION
**Pull request by srfraser**
This splits up the workload for the data collection into three parts:
1. A scanner, which finds the taskgraphs that need working on, from either the pushlog or the nightly index. It then calls the graph analyzer for each one.
2. A graph analyzer, which takes a data record and a task graph id, and fills in the missing bits of data. This writes output to a staging area.
3.  A periodic data collator, which scans the staging area and concatenates all the files there into one.

The AWS database scanner we're using should be fine with the split files, and the data should still appear as soon as it's present, but the collator is good for keeping the number of files under control and making the queries more efficient.

This is all stored under `v4` of the data, so we'll need to rerun the AWS Glue and do some work to update the graphs. We should be able to copy over the old data, it'll just be missing the artifact sizes.

### CODE REVIEW

File: measuring_ci/artifacts.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            aws_access_key_id=os.environ.get('TASKCLUSTER_S3_ACCESS_KEY'),
+            aws_secret_access_key=os.environ.get('TASKCLUSTER_S3_SECRET_KEY'),
+        )
+    loop = asyncio.get_event_loop()
+    log.debug('Fetching S3 artifact info for %s', taskid)
+    artifacts = []
+
+    cont_token = None
+    prefix = taskid + '/'
+    while True:
```
Comment (catlee) [inline measuring_ci/artifacts.py]: this code is common between this function and utils.py:find_staged_data_files. maybe we can factor out the s3 logic here?

File: parquet_collator.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    # Don't store the full path to integration/releases/...
+    if 'project' in args:
+        short_project = args['project'].split('/')[-1]
+        config['total_cost_output'] = config['total_cost_output'].format(project=short_project)
+        config['staging_output'] = config['staging_output'].format(project=args['project'])
+    staged_files = await find_staged_data_files(config['staging_output'])
+
+    # Without this, urllib doesn't know about s3 and removes s3://netloc/
+    urllib.parse.uses_netloc.append('s3')
+    urllib.parse.uses_relative.append('s3')
```
Comment (catlee) [inline parquet_collator.py]: it looks like these affect the global urllib.parse state? maybe they should be moved up into the top-level module?

File: measuring_ci/artifacts.py [hunk 3]
```diff
@@ -47,29 +46,9 @@
...
-        resp = await loop.run_in_executor(None, func)
-        if resp['KeyCount'] == 0:
-            break
-        artifacts.extend(resp['Contents'])
-        if not resp['IsTruncated']:
-            break
-        cont_token = resp['NextContinuationToken']
-
-    return artifacts
+    return list_s3_objects(s3_client, bucket_name, prefix)
```
Comment (catlee) [inline measuring_ci/artifacts.py]: I think you need an await here and below in find_staged_data_files

Comment (srfraser) [reply measuring_ci/artifacts.py]: Thanks!

**Pull request merged**